File: native_library/apple/FirebaseANE/VisionFaceANE/Extensions/FreVisionPoint.swift
/*
 *  Copyright 2018 Tua Rua Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import Foundation
import FreSwift
import FirebaseMLVision

public extension VisionPoint {
    func toFREObject() -> FREObject? {
        do {
            let ret = try FREObject(className: "com.tuarua.firebase.vision.Point")
            try ret?.setProp(name: "x", value: self.x)
            try ret?.setProp(name: "y", value: self.y)
            try ret?.setProp(name: "z", value: self.z)
            return ret
        } catch {
        }
        return nil
    }
}

File: native_library/android/FirebaseANE/Vision/src/main/java/com/tuarua/firebase/vision/extensions/FreVisionPoint.kt
package com.tuarua.firebase.vision.extensions

import com.adobe.fre.FREObject
import com.google.firebase.ml.vision.common.FirebaseVisionPoint
import com.tuarua.frekotlin.FREObject
import com.tuarua.frekotlin.FreException

fun FirebaseVisionPoint.toFREObject(): FREObject? {
    try {
        return FREObject("com.tuarua.firebase.vision.VisionPoint", x, y, z ?: 0)
    } catch (e: FreException) {
    }
    return null
}